### PR TITLE
Fix batch_sampler maybe None error

### PR DESCRIPTION
**Pull request by candlewill**
# Fix batch_sampler maybe None error

Fixes # (issue)

https://github.com/huggingface/accelerate/issues/3011

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/accelerate/blob/main/CONTRIBUTING.md#submitting-a-pull-request-pr),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/accelerate/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/accelerate/tree/main/docs#writing-documentation---specification).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

- Core parts of the library: @BenjaminBossan

### CODE REVIEW

**Comment by BenjaminBossan** [comment]
Thanks a lot for creating the PR. If you could also add a unit test to [`test_data_loader.py`](https://github.com/huggingface/accelerate/blob/main/tests/test_accelerator.py), that would be great. The test could basically be the same as the example you provided in the issue.

**Review by muellerzr — COMMENTED** [review]
agreed with Benjamin, let's add a test here then ✅ , thanks!

**Comment by candlewill** [comment]
The unit test code is added, please help to review and merge.

File: tests/test_data_loader.py [hunk 1]
```diff
@@ -364,6 +380,14 @@ def test_iterable_dataset_shard(self):
         self.check_iterable_dataset_shards(dataset, seed, batch_size=4, drop_last=False, split_batches=True)
         self.check_iterable_dataset_shards(dataset, seed, batch_size=4, drop_last=True, split_batches=True)
 
+    def test_iterable_dataset_using_none_batch_size():
```
Comment (muellerzr) [inline tests/test_data_loader.py:383]: This is part of a `unittest.TestCase`, so it needs `self`
```suggestion
    def test_iterable_dataset_using_none_batch_size(self):
```

**Review by muellerzr — APPROVED** [review]
Thanks! Overall this looks good to me, please run `pip install -e .[quality]; make style; make quality;` to fix the quality failures, and I made a comment for why the tests are failing

**Review by BenjaminBossan — CHANGES_REQUESTED** [review]
Thanks for adding the test @candlewill. This PR is almost good to go, only the linter needs to be satisfied. This is because [these lines](https://github.com/huggingface/accelerate/pull/3025/files#diff-09bbe4f689c597ef1e168657171d49d91220c431c79f98983603ad7bc7551cb0R49-R60) are indented one level too far.

**Comment by candlewill** [comment]
I am sure now everything is OK, please help review again.

**Review by muellerzr — APPROVED** [review]
Thanks for adding a test!

**Pull request merged**